### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by arbitrary-CodeBeholder**

### CODE REVIEW

File: new-contract-interaction/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "description": "Forta Agent template for monitoring new contracts interacting with {protocolName} contracts",
+    "scripts": {
+      "start": "npm run start:dev",
+      "start:dev": "nodemon --watch src --watch forta.config.json --watch agent-config.json -e js,json --exec 'forta-agent run'",
+      "start:prod": "forta-agent run --prod",
+      "tx": "forta-agent run --tx",
+      "block": "forta-agent run --block",
+      "range": "forta-agent run --range",
+      "file": "forta-agent run --file",
+      "publish": "forta-agent publish",
```
Comment (arbitrary-sparrow) [inline new-contract-interaction/package.json:13]:
```suggestion
      "publish": "forta-agent publish",
      "push": "forta-agent push",
```

File: new-contract-interaction/SETUP.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# New Contract Interaction Agent Template
+
+This agent monitors blockchain transactions for new contracts interacting with specific contract
+addresses. Alert type and severity are specified per contract.
```
Comment (echoAE) [inline new-contract-interaction/SETUP.md:4]:
```suggestion
This agent monitors blockchain transactions for new contracts and/or EOAs with few transactions
interacting with specific contract addresses. Alert type and severity are specified per contract.
```

File: new-contract-interaction/agent-config.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (arbitrary-sparrow) [inline new-contract-interaction/agent-config.json:1]: I don't think we need this file since we have the `agent-config.json.example` file

Comment (echoAE) [inline new-contract-interaction/agent-config.json:1]: This file can be removed from the repository.  We are using the `agent-config.json.example` file as the template for the user to copy and modify.

File: new-contract-interaction/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "forta-{developerAbbreviation}-{protocolAbbreviation}-contract-interaction",
+    "version": "0.0.1",
+    "description": "Forta Agent template for monitoring new contracts interacting with {protocolName} contracts",
```
Comment (echoAE) [inline new-contract-interaction/package.json:4]:
```suggestion
    "description": "Forta Agent template for monitoring new contracts and EOAs interacting with {protocolName} contracts",
```

File: new-contract-interaction/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "forta-{developerAbbreviation}-{protocolAbbreviation}-contract-interaction",
```
Comment (echoAE) [inline new-contract-interaction/package.json:2]:
```suggestion
    "name": "forta-{developerAbbreviation}-{protocolAbbreviation}-new-contract-eoa-interaction",
```

File: new-contract-interaction/package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      "file": "forta-agent run --file",
+      "publish": "forta-agent publish",
+      "push": "forta-agent push",
+      "disable": "forta-agent disable",
+      "enable": "forta-agent enable",
+      "keyfile": "forta-agent keyfile",
+      "test": "jest"
+    },
+    "dependencies": {
+      "axios": "^0.21.4",
```
Comment (echoAE) [inline new-contract-interaction/package.json:21]:
```suggestion
```

File: new-contract-interaction/package.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      "disable": "forta-agent disable",
+      "enable": "forta-agent enable",
+      "keyfile": "forta-agent keyfile",
+      "test": "jest"
+    },
+    "dependencies": {
+      "axios": "^0.21.4",    
+      "bignumber.js": "^9.0.2",
+      "forta-agent": "^0.0.33",
+      "dotenv": "^10.0.0"
```
Comment (echoAE) [inline new-contract-interaction/package.json:24]:
```suggestion
```

File: new-contract-interaction/package.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      "publish": "forta-agent publish",
+      "push": "forta-agent push",
+      "disable": "forta-agent disable",
+      "enable": "forta-agent enable",
+      "keyfile": "forta-agent keyfile",
+      "test": "jest"
+    },
+    "dependencies": {
+      "axios": "^0.21.4",    
+      "bignumber.js": "^9.0.2",
```
Comment (echoAE) [inline new-contract-interaction/package.json:22]:
```suggestion
```

File: new-contract-interaction/package.json (hunk 7 above)
Comment (echoAE) [inline new-contract-interaction/package.json:24]:
```suggestion
      "forta-agent": "^0.0.36"
```

File: new-contract-interaction/SETUP.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+```
+  new-contract-interaction/
+    README.md
+    SETUP.md
+    COPYING
+    LICENSE
+    Dockerfile
+    forta.config.json
+    package.json
+    agent-config.json
```
Comment (echoAE) [inline new-contract-interaction/SETUP.md:68]: These files need to be added to the repository:
```
COPYING
LICENSE
Dockerfile
```

File: new-contract-interaction/src/agent.js [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+  findingType,
+  findingSeverity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation
+) {
+  const finding = {
+    name: `${protocolName} Contract Interaction`,
+    description: `The ${contractName} contract interacted with a new contract ${interaction_address}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-CONTRACT-INTERACTION`,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:25]:
```suggestion
    alertId: `${developerAbbreviation}-${protocolAbbreviation}-NEW-CONTRACT-INTERACTION`,
```

File: new-contract-interaction/SETUP.md (hunk 9 above)
Comment (arbitrary-sparrow) [reply new-contract-interaction/SETUP.md:68]: The `README.md` in the base of the repository needs to be updated to include this agent as well.

File: new-contract-interaction/src/agent.js [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+const config = require('../agent-config.json');
+
+// set up a variable to hold initialization data used in the handler
+const initializeData = {};
+
+// helper function to create alert for contract interaction
+function createContractInteractionAlert(
+  contractName,
+  contractAddress,
+  interaction_address,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:14]:
```suggestion
  interactionAddress,
```

File: new-contract-interaction/src/agent.js [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+    name: `${protocolName} Contract Interaction`,
+    description: `The ${contractName} contract interacted with a new contract ${interaction_address}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-CONTRACT-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interaction_address,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:32]:
```suggestion
      interactionAddress,
```

File: new-contract-interaction/src/agent.js [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+
+  return Finding.fromObject(finding);
+}
+
+// helper function to create alert for EOA interaction
+function createEOAInteractionAlert(
+  contractName,
+  contractAddress,
+  interaction_address,
+  transaction_count,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:45]:
```suggestion
  interactionAddress,
  transactionCount,
```

File: new-contract-interaction/src/agent.js [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+  blockNumber,
+  findingType,
+  findingSeverity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation
+) {
+  const finding = {
+    name: `${protocolName} Contract Interaction`,
+    description: `The ${contractName} contract interacted with a new contract ${interaction_address}`,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:24]:
```suggestion
    description: `The ${contractName} contract interacted with a new contract ${interactionAddress}`,
```

File: new-contract-interaction/src/agent.js [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+  transaction_count,
+  findingType,
+  findingSeverity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation
+) {
+  const finding = {
+    name: `${protocolName} EOA Interaction`,
+    description: `The ${contractName} contract interacted with a new EOA ${interaction_address}`,
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:54]:
```suggestion
    description: `The ${contractName} contract interacted with a new EOA ${interactionAddress}`,
```

File: new-contract-interaction/src/agent.js [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-EOA-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interaction_address,
+      transaction_count
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:63]:
```suggestion
      interactionAddress,
      transactionCount,
```

File: new-contract-interaction/src/agent.js [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+    description: `The ${contractName} contract interacted with a new contract ${interaction_address}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-CONTRACT-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interaction_address,
+      blockNumber
```
Comment (echoAE) [inline new-contract-interaction/src/agent.js:33]:
```suggestion
      blockNumber,
```

File: new-contract-interaction/src/agent.js [hunk 18]
```diff
@@ -0,0 +1,218 @@
...
+    description: `The ${contractName} contract interacted with a new contract ${interactionAddress}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-NEW-CONTRACT-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interactionAddress,
+      blockNumber,
```
Comment (arbitrary-sparrow) [inline new-contract-interaction/src/agent.js:33]: Transaction and block information is provided by forta explorer so we don't need to add `blockNumber` to the metadata

**Review by echoAE — APPROVED** [review]

**Review by arbitrary-sparrow — APPROVED** [review]

**Pull request merged**